### PR TITLE
BGL descriptor with graph

### PR DESCRIPTION
**Pull request by maxGimeno**
This PR adds the class Graph_with_descriptor_with_graph to the package BGL.
 This class is a wrapper for a Graph that allow its descriptor to know their graph. It is mainly for debugging purposes.

Small feature [page](https://cgal.geometryfactory.com/CGAL/Members/wiki/Features/Small_Features/Graph_with_descriptor_with_graph)

Note: small feature pre-approved by Sloriot 20 December 2016

### CODE REVIEW

**Comment by lrineau** [comment]
This PR needs a small feature, so that CGAL developers can have a chance to comment on the added documentation.

File: Installation/changes.html [hunk 1]
```diff
@@ -219,6 +219,10 @@ <h2 id="release4.10">Release 4.10 </h2>
     convert a <code>Nef_polyhedron_3</code> to any model of the <code>MutableFaceGraph</code>
     concept.
   </li>
+  <li>
+    Add class <code>CGAL::Graph_with_descriptor_with_graph</code> to wrap an existing 
+    Graph and provide a reference to the said Graph to all of its descriptors.
+  </li>
```
Comment (sloriot) [inline Installation/changes.html]: that wraps ... and provides?

File: BGL/include/CGAL/boost/graph/Gwdwg.h [hunk 2]
```diff
@@ -0,0 +1,735 @@
...
+The class `Graph_with_descriptor_with_graph` wraps a graph into another graph in such a way that its descriptors contain a reference to the graph they come from.
+
+For example, calling `source(edge, graph)` will trigger an assertion if `edge` does not belong to `graph`.
+It is mainly used for debugging purposes.
+
+Property forwarding
+-------------------
+All internal properties of the underlying graph are forwarded.
+
+Property maps can be wrapped with Graph_with_descriptor_with_graph_property_map.
```
Comment (sloriot) [inline BGL/include/CGAL/boost/graph/Gwdwg.h]: backticks missing

File: BGL/include/CGAL/boost/graph/Gwdwg.h [hunk 3]
```diff
@@ -0,0 +1,735 @@
...
+\tparam Graph must be a model of a `FaceListGraph` and a `HalfedgeListGraph`.
```
Comment (sloriot) [inline BGL/include/CGAL/boost/graph/Gwdwg.h]: a model of `FaceListGraph` and `HalfedgeListGraph`

File: Installation/changes.html (hunk 1 above)
Comment (afabri) [inline Installation/changes.html]: also the word graph should be in lower case as it is just a word.

**Comment by maxGimeno** [comment]
Warnings fixed in test.

**Pull request merged**